docs/kilnjs.md

```diff
@@ -0,0 +1,220 @@
+
+  subscriptions.subscribe('UPDATE_COMPONENT', (payload)=> {
+    // do something when a component has been updated
+    // you can also test the type of component updated
+    // by checking the value of payload.data.componentVariation
+    // and only reacting when it's a certain type of component
+    // or if the component's new value is equal or not to some value
+  }, false);
+
+  schema.myDate.subscribe('UPDATE_COMPONENT', (payload)=> {
+    // same as above, just attached to the myDate field, which implies
+    // you want to do something to the myDate field with the payload
+  });
+
+  return schema;
+};
+
+```
+
+---
+
+## Kilnjs custom methods
+
+KilnInput also provides its own set of custom methods.
+
+### getComponentInstances
+
+* ***getComponentInstances(componentName)*** - returns an array containing the uris of all components of type componentName that are on the current page.
+
+```
+kilnInput.getComponentInstances('paragraph');
+```
+
+### reRenderInstance
+
+* ***reRenderInstance(uri)*** - Fetchs the component's data and passes it through the components model.render function, thus refreshing it on the page. For instance, if some outside source has updated component data and does not trigger a page refresh, you could force a component rerender that would include the updated data.
+
+```
+// if a paragraph component is updated, then rerender the instances of the snash component
+kilnInput.subscribe('UPDATE_COMPONENT', (payload)=> {
+    if (payload.data.componentVariation === 'paragraph') {
+      let snashInstances = eventBus.getComponentInstances('snash');
+      snashInstances.forEach((instance) => {
+        kilnInput.reRenderInstance(instance);
+      });
+    }
+  }, false);
+```
+
+### saveComponent
+
+* ***saveComponent(uri, data)*** - Saves the component at the provided uri with the provided data.  Saving a component this way will also rerender the it on the page with the updated data.
+
+```
+const uri = 'localhost/_components/paragraph/instances/cjue4dl7i00062a65rpexptmp',
+  data = { text: "This is a paragraph" };
+
+kilnInput.saveComponent(uri, data);
+```
+
+
+
+### setProp
+
+* ***setProp(prop, value)*** - change value of a property on the input
+
+```
+kilninput.setProp('_has', { ...kilninput['_has'], input: 'select' });
+```
+
+### show/hide
+
+* ***show()*** - used to make an input visible.
+
+* ***hide()*** - used to make an input invisible
+
+```
+  schema.enableSocialButtons.on('input', (val) => {
+    if(val) {
+      schema.shareServices.show();
+    } else {
+      schema.shareServices.hide();
+    }
+  });
+```
+
+### url
+
+* ***url()*** - returns the url object from Vuex state of the component that is currently being edited. Object contains the component name, the instance, and the path
+
+```
+  kilnInput.url()
+
+  /// returns an object structured like this
+
+  {
+    component: "meta-title"
+    instance: "cjtfuc3rw00019fz9egagqev0"
+    path: "settings"
+  }
```

Comment:
I feel that it would be better if we put this whole block inside a comment block
```js
kilnInput.url()

/**
 * returns an object structured like this
 *
 * {
 *   component: "meta-title"
 *   instance: "cjtfuc3rw00019fz9egagqev0"
 *   path: "settings"
 * }
 */
```

Reply:
👍